test/unit/nics/bmc_test.rb

```diff
@@ -5,4 +5,45 @@ class BMCTest < ActiveSupport::TestCase
     host = FactoryGirl.build(:host, :managed)
     assert FactoryGirl.build(:nic_bmc, :host => host, :provider => 'ipmi').valid?
   end
+
+  context "no BMC smart proxy exists" do
+    def setup
+      SmartProxy.with_features('BMC').destroy_all
+    end
+
+    test 'requires BMC proxy without subnet' do
+      host = FactoryGirl.build(:host, :managed)
+      bmc_nic = FactoryGirl.build(:nic_bmc, :host => host)
+      host.interfaces << bmc_nic
+      refute_with_errors bmc_nic.valid?, bmc_nic, :base, /no proxy/
+    end
+
+    test 'requires BMC proxy in the same subnet' do
+      host = FactoryGirl.build(:host, :managed)
+      subnet = FactoryGirl.build(
+        :subnet,
+        :tftp => nil,
```

Comment:
It looks like the default associations of :tftp and :dhcp should be removed from the factory, which would allow this and other tests simply to use a sensible default of an empty subnet sans associations.

The only thing it breaks is `DhcpOrchestrationTest#test_0006_provision interface DHCP records should contain filename/next-server attributes` which tries to create a subnet via the host factory with both, which won't work (as one trait overwrites the other), so we could move the slight ugliness to that test instead, and improve the factory defaults.